src/diagnostic-channel-publishers/src/pg-pool.pub.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+import {EventEmitter} from "events";
+
+function postgresPool1PatchFunction(originalPgPool) {
+    const originalConnect = originalPgPool.prototype.connect;
+
+    originalPgPool.prototype.connect = function connect(callback?: Function): void {
+        callback = callback
+            ? channel.bindToContext(callback)
```

Comment:
Do you think this unwrap-rewrap should just be part of the `bindToContext` API? I feel like this would be a concern nearly anywhere. For what its worth, it does currently work, at least when the context preserving function is Zone.wrap, but it might not if the context preserver has side effects.